### PR TITLE
add natGateways tasks for ALICloud

### PR DESCRIPTION
**Pull request by LilyFaFa**
Add natGateway tasks for AliCloud
This is part of #4127.

### CODE REVIEW

**Comment by chrisz100** [comment]
Not an ali expert, but code-wise
/lgtm

File: pkg/resources/ali/ali.go [hunk 1]
```diff
@@ -689,3 +693,138 @@ func DeleteVolume(cloud fi.Cloud, r *resources.Resource) error {
 	}
 	return nil
 }
+
+func ListNatGateway(d *clusterDiscoveryALI, vpcID string, resourceTrackers *[]*resources.Resource) error {
+
+	// Delete NateGateway with specified name. All of the EIPs will be deleted.
```
Comment (justinsb) [inline pkg/resources/ali/ali.go]: Any tagging you can do to be more confident is great, but at least on AWS & GCP not all objects can be tagged.  (But I think on AWS we're down to just SSH keys not being taggable!)

**Comment by justinsb** [comment]
Looks great - thanks @LilyFaFa 

/approve
/lgtm

**Pull request merged**